BeeJet.Bot/Commands/Handlers/GameManagement/AddGameCommandHandler.cs

```diff
@@ -25,25 +25,25 @@ public void RegisterOptions(SlashCommandBuilder builder)
         }
 
         [BeeJetBotSlashCommand("add-game", "Add game channel", nameof(RegisterOptions))]
-        public async Task SlashCommandExecuted(SlashCommandContext context)
+        public async Task SlashCommandExecuted()
         {
-            var gameName = (string)context.DiscordContext.Data.Options.First().Value;
+            var gameName = (string)Context.SlashCommandInteraction.Data.Options.First().Value;
```

Comment:
Zit hier over te twijfelen.
Dus heel verhaal met mijn "waarom"

Ik zie namelijk elke keer Context.SlashCommandInteraction.
Dus eigenlijk wil je veelal iets met de SlashCommandInteraction en niet zozeer met de Context.
Zou je dan niet vanuit hier direct met de SlashCommandInteraction willen werken?
Of zou de context deze functies niet moeten hebben zodat je niet op meer niveaus hoeft door te punten?

Reden waarom ik dit zeg zijn weer die mooie Clean Code en SOLID principes. 
Wanneer je doorpunt, betekend dat je van buitenaf de implementatie moet snappen (Kortom de Context en SlashCommandInteraction is dan blijkbaar niet specifiek genoeg)
Je zou dus bijvoorbeeld in de Context iets kunnen hebben als DetermineGame en DetermineCategory.
Of misschien Context.SlashCommandInteraction.DetermineGame (al is dit ook al doorgepunt).

Reply:
> Zou je dan niet vanuit hier direct met de SlashCommandInteraction willen werken?
Zelfde verhaal als die andere je wilt die context hebben, zodat je straks generieke services kan aanspreken. Daarvoor heb je een custom context nodig, die je zelf kan beheren. Dus daar kunnen we niet iets gebruiken uit de discord api.

>Of zou de context deze functies niet moeten hebben zodat je niet op meer niveaus hoeft door te punten?
Dat lijkt me niet, want dan moet je alles gaan doorlussen, lijkt me wat teveel van het goeie

>Je zou dus bijvoorbeeld in de Context iets kunnen hebben als DetermineGame en DetermineCategory.
De context is in dit geval de interactie vanuit discord, net zoals bv bij een .net web api er een httpcontext is. Je code maakt er gebruik van, maar de context zelf zou geen logica moeten bevatten. Dus lijkt mij dat in een context geen determinegame of category thuis horen, want die interpreteren de context.

Reply:
Mijn punt zat hem meer in dat je veelal Context.SlashCommandInteraction.<Vervolgmethode> weer terug ziet komen en hiervan weer verschillende zaken. Dat was mijn reden om de Context dan de mogelijkheden te geven om hier gegevens uit te vragen.
Je lust het inderdaad dan door. 
Maar ook met een goede naam :)

Context.SlashCommandInteraction.Data.Options.First().Value;  <- dit zegt net dat je een GameName wilt hebben
Context.GetGameName die dan SlashCommandInteraction.Data.Options.First().Value doet, geeft dat wel direct aan :)

Zelfde met bepalen CategoryName.
Context.SlashCommandInteraction.Data.Options.FirstOrDefault(b => b.Name == "category");
of
Context.GetCategoryName welke 
            var category = Context.SlashCommandInteraction.Data.Options.FirstOrDefault(b => b.Name == "category");
            if (category != null)
            {
                categoryName = (string)category.Value;

            }
dan doet voor je.

Reply:
Maar dat gaat niet werken, want context is iets algemeens, wat alle commandhandlers krijgen. Of het nu een sync-steam is of zoals hier de add-game. Dus daar wil je niet dat soort methodes op hebben, die specifiek voor 1 command zijn.

ik zou dan eerder op de AddGameCommandHandler een functie toevoegen en niet op de SlashCommandContext
`
private string GetCategoryName()
        {
            var categoryName = "Gaming";//Default name
            var category = Context.SlashCommandInteraction.Data.Options.FirstOrDefault(commandOptionData => commandOptionData.Name == "category");
            if (category != null)
            {
                categoryName = (string)category.Value;
            }

            return categoryName;
        }
`



@pyrocumulus / @MauricePonte  / @GeKopt  : denk dat de grote vraag is, gaan we de hele discord api uit onze code wegwerken door alles te wrappen? Persoonlijk zou ik het jammer vinden, want dat kost klauwen aan tijd. 
Dan gaan we de kant op dat alle klasses, IGuild, IChannel, ISlashCommandInteraction en alle andere klasses waar je actie op uitvoert een manager klasse moet hebben, die alles gaat doorlussen naar exact dezelfde methode op de interface van discord. Zie zelf daar niet een nut in. Als we extra gedrag willen hebben op een IGUild die herbruikbaar moet zijn, dan lijkt me dat iets voor een extension method.

Reply:
An sich leek me die `IGuildManager` wel okay als proxy idee. Maar als we dat inderdaad voor alle Discord api types moeten doen dan denk ik dat we iets te ver doorschieten, want dat wordt wel een werk en het maakt het er denk ik niet duidelijker op. We kunnen toch tegen niets anders dan Discord praten dus daar zou het niet bij helpen en dan blijft enkel testbaarheid over.

Ik zou denk ik enkel proberen te wrappen als we niet een interface hebben die we kunnen substituten. Dus puur de managers etc. toevoegen als het niet anders kan vanwege testbaarheid.

Reply:
Het zou kunnen maar dan moeten we wel overal met de interfaces van Discord werken en niet ineens halverwege naar implementatie klasses schieten (want dan krijg je issues met testen).

Extension methodes voor het uitbreiden of versimpelen kan ook een optie zijn. 
Moet even kijken waar ik dan tegenaan loop met unit testen.